client/src/components/Header/components/nav-links.tsx

```diff
@@ -16,33 +16,43 @@ import {
   faExternalLinkAlt
 } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import React, { Component, Fragment } from 'react';
+import React, { Component, Fragment, createRef, Ref } from 'react';
 import { TFunction, withTranslation } from 'react-i18next';
 import { connect } from 'react-redux';
 import envData from '../../../../../config/env.json';
 import {
   availableLangs,
-  langDisplayNames
+  langDisplayNames,
+  langCodes
 } from '../../../../../config/i18n/all-langs';
 import { hardGoTo as navigate } from '../../../redux';
 import { updateUserFlag } from '../../../redux/settings';
 import createLanguageRedirect from '../../create-language-redirect';
 import { Link } from '../../helpers';
 import { Themes } from '../../settings/theme';
+import LanguageGlobe from '../../../assets/icons/language-globe';
 
 const { clientLocale, radioLocation, apiLocation } = envData;
 
 const locales = availableLangs.client;
 
+const languageGlobeIconStyle = {
+  height: '1rem',
+  width: '1rem'
+};
```

Comment:
We use LanguageGlobe once, right?  If so, we can just fold this into the component.

Reply:
> If I tab outside the menu it closes. Is this necessary? It seems like that would be an issue if a user accidentally presses tab too often.

Ya, this is one of those "gray" areas when it comes to accessibility. The nav menu is technically a "disclosure" widget. The W3C ARIA Authoring Practices actually has an example of a [disclosure pattern for navigation menus](https://w3c.github.io/aria-practices/examples/disclosure/disclosure-navigation.html). If you look at note 3 under "Accessibility Features" it says:

_If a dropdown is open and focus is inside the navigation region, pressing Esc will close the dropdown. **Moving focus out of the navigation region also closes an open dropdown.**_

That's the pattern I was following here but I can completely understand reasons for not doing this. I think one of the main reasons I lean toward autoclose is because with narrower view ports (narrow enough that it shows the tab interface) the dropdown menu covers the tabs completely, and at some point the menu covers the entire top half of the page (or more, depending on the view port height). So that could potentially lead to users losing sight of the focus indicator because it is hidden underneath the menu, which I think an argument could be made that it would then violate [WCAG SC 2.4.7](https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_customize&levels=aaa#focus-visible). 

A possible compromise? The next tab stop after the last menu item is the profile icon. We could keep the menu expanded if the user tabs to that icon, which would allow them one "free" tab, and then close the menu if they tab off of the icon into the content.

Reply:
That's fair enough.  It's a little annoying to have to shift-tab back and try to navigate again if you accidentally go too far, but it's not *that* bad.  I think it's fine to err on the side of following the w3c examples.

We could go with the compromise, but the code is already a little on the complicated side. I'd rather keep it simple for now.  If we get feedback from users that this is frustrating, we can revisit this, but it seems good for now.

Reply:
Leaving as is for now.